### PR TITLE
Make map entrance/exit fetching functions use Coords objects

### PR DESCRIPTION
**Pull request by tupaschoal**

### CODE REVIEW

File: src/openrct2/actions/PlaceParkEntranceAction.hpp [hunk 1]
```diff
@@ -79,13 +79,12 @@ DEFINE_GAME_ACTION(PlaceParkEntranceAction, GAME_COMMAND_PLACE_PARK_ENTRANCE, Ga
...
         for (uint8_t index = 0; index < 3; index++)
         {
             if (index == 1)
             {
-                entranceLoc.x += CoordsDirectionDelta[(_loc.direction - 1) & 0x3].x;
-                entranceLoc.y += CoordsDirectionDelta[(_loc.direction - 1) & 0x3].y;
+                entranceLoc += CoordsDirectionDelta[(_loc.direction - 1) & 0x3];
             }
             else if (index == 2)
             {
```
Comment (duncanspumpkin) [inline src/openrct2/actions/PlaceParkEntranceAction.hpp]: You can change L91 as well

File: src/openrct2/world/Map.cpp [hunk 2]
```diff
@@ -1790,17 +1791,17 @@ EntranceElement* map_get_park_entrance_element_at(int32_t x, int32_t y, int32_t
     return nullptr;
 }
 
-EntranceElement* map_get_ride_entrance_element_at(int32_t x, int32_t y, int32_t z, bool ghost)
+EntranceElement* map_get_ride_entrance_element_at(const TileCoordsXYZ& entranceTileCoords, bool ghost)
```
Comment (duncanspumpkin) [inline src/openrct2/world/Map.cpp]: Make it `CoordsXYZ`

File: src/openrct2/world/Map.cpp [hunk 3]
```diff
@@ -1815,17 +1816,17 @@ EntranceElement* map_get_ride_entrance_element_at(int32_t x, int32_t y, int32_t
     return nullptr;
 }
 
-EntranceElement* map_get_ride_exit_element_at(int32_t x, int32_t y, int32_t z, bool ghost)
+EntranceElement* map_get_ride_exit_element_at(const TileCoordsXYZ& exitTileCoords, bool ghost)
```
Comment (duncanspumpkin) [inline src/openrct2/world/Map.cpp]: Make it `CoordsXYZ`

File: src/openrct2/actions/PlaceParkEntranceAction.hpp (hunk 1 above)
Comment (tupaschoal) [reply src/openrct2/actions/PlaceParkEntranceAction.hpp]: I can't there is a multiplication first, only if I implement `operator*`

**Comment by tupaschoal** [comment]
> Change it so that the types of these functions generally are `CoordsXYZ`. I don't see this as being anywhere near as special as get surface

Personally I think it looks uglier, people have to remember to do the conversions, but if the interface should always get a `CoordsXY`, it makes sense.

**Comment by Gymnasiast** [comment]
@tupaschoal We're generally moving towards using `CoordsXY(ZDE) for everything. At some point, the ugliness should go away.

However, for now, I do still think there has to be a better way than calling `map_get_entrance_exit_element_at()` with bitshifts. Perhaps you could add another conversion method to one of the coordinate objects?

**Comment by tupaschoal** [comment]
Ok, I've created a helper function for them to convert into each other.

**Pull request merged**